apps/prairielearn/src/migrations/20220905173159__rubrics__create.sql

```diff
@@ -0,0 +1,60 @@
+CREATE TABLE IF NOT EXISTS
+  rubrics (
+    id BIGSERIAL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_extra_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL,
+    replace_auto_points BOOLEAN NOT NULL,
+    modified_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    deleted_at TIMESTAMP WITH TIME ZONE
+  );
+
+CREATE TABLE IF NOT EXISTS
+  rubric_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    number BIGINT NOT NULL,
+    points DOUBLE PRECISION NOT NULL,
+    description VARCHAR(100) NOT NULL,
+    explanation VARCHAR(10000),
+    grader_note VARCHAR(10000),
+    key_binding VARCHAR(30),
+    deleted_at TIMESTAMP WITH TIME ZONE
+  );
+
+CREATE INDEX IF NOT EXISTS rubric_items_rubric_id ON rubric_items (rubric_id);
+
+CREATE TABLE IF NOT EXISTS
+  rubric_gradings (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    computed_points DOUBLE PRECISION NOT NULL,
+    adjust_points DOUBLE PRECISION NOT NULL,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_extra_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL
+  );
+
+CREATE TABLE IF NOT EXISTS
+  rubric_grading_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_grading_id BIGINT NOT NULL REFERENCES rubric_gradings (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    rubric_item_id BIGINT NOT NULL REFERENCES rubric_items (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    score DOUBLE PRECISION NOT NULL DEFAULT 1,
+    points DOUBLE PRECISION NOT NULL,
+    description VARCHAR(100) NOT NULL,
+    note VARCHAR(10000)
+  );
+
+ALTER TABLE rubric_grading_items
+ADD CONSTRAINT rubric_grading_items_rubric_grading_id_rubric_item_id_key UNIQUE (rubric_grading_id, rubric_item_id);
```

Comment:
Not that it really matters, but you can write these UNIQUE constraints inside the CREATE TABLE expressions by writing `UNQIUE (...)` as the last line in the table definition.

Reply:
Updated in 8badea102.